### PR TITLE
Check if parquet is installed as a proxy for fully-functional arrow

### LINKED ISSUE #2091: Solaris error on CRAN
I noticed that there's a Solaris error on CRAN for the Arrow-DuckDB integration tests.

One of the ways that we gate tests is by turning them on/off [if the features they use are available](https://github.com/apache/arrow/blob/master/r/tests/testthat/helper-skip.R#L18-L34)

Adding something like the following to the top of the arrow integration tests should accomplish this (after line 7 or so):

```
skip_if_not("dataset" %in% arrow::arrow_info()$capabilities)
```

Which will skip if dataset support has not been built for Arrow (which it almost certainly will not be on Solaris!). Datasets isn't the _only_ feature one could test, but since that is one of the big use-cases it might be a good one to turn these tests off with.

### PR DESCRIPTION
**Pull request by jonkeane**
resolves #2091

### CODE REVIEW

**Comment by jonkeane** [comment]
I also moved the skipping up in the file so we only have to do it once, but I'm happy to move it back in to each of the testthat blocks if you prefer it there.

File: tools/rpkg/tests/testthat/test_fetch_arrow.R [hunk 1]
```diff
@@ -1,17 +1,19 @@
 library("testthat")
 library("DBI")
 
+skip_on_os("windows")
+skip_if_not_installed("arrow", "4.0.1")
+# Skip if parquet is not a capabiltiy as an indicator that Arrow is fully installed.
+full_arrow <- "parquet" %in% names(arrow::arrow_info()$capabilities) && arrow::arrow_info()$capabilities["parquet"]
+skip_if_not(full_arrow, message = "The installed Arrow is not fully featured, skipping Arrow integration tests")
```
Comment (nealrichardson) [inline tools/rpkg/tests/testthat/test_fetch_arrow.R:8]:
```suggestion
skip_if_not(arrow::arrow_with_parquet(), message = "The installed Arrow is not fully featured, skipping Arrow integration tests")
```

Comment (jonkeane) [reply tools/rpkg/tests/testthat/test_fetch_arrow.R:8]: Ah of course, I forgot about this helper! Thanks

**Comment by Mytherin** [comment]
To me this looks like it is ready to merge, but there appears to be a merge conflict. Could you have perhaps having a look at merging with master? After that I think it is good to go.

**Comment by Mytherin** [comment]
> To me this looks like it is ready to merge, but there appears to be a merge conflict. Could you perhaps have a look at merging with master? After that I think it is good to go. Thanks!

Actually not required anymore, @pdet helped me resolve them. Should be fixed now.

**Pull request merged**